### ISSUE TITLE
Add a plugin for Klaviyo

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#4885]
### Feature description

Details about the needed integration TBD https://developers.klaviyo.com/en/reference/api_overview

**Comment by dipeshsingh253** [comment]
Hi @anna-geller , I would like to contribute to this issue. Can you please assign it to me??

**Comment by dipeshsingh253** [comment]
Hi @anna-geller , I have set up the dev env locally and I have gone through the [plugin-dev-guide](https://developers.klaviyo.com/en/reference/api_overview).

Can you explain what exactly we have to implement here?? Because this [klaviyo](https://developers.klaviyo.com/en/reference/api_overview) docs seems to be a general dev doc for Klaviyo.

**Comment by Ben8t** [comment]
@dipeshsingh253 Klaviyo has several API endpoints. I think a good start would be to implement the [Campaign API ](https://developers.klaviyo.com/en/reference/campaigns_api_overview)(Jobs, Campaigns, Messages, Relationships) and see how it goes :)

@anna-geller I'm not very familiar with Klaviyo, do we have any prior incentive regarding this plugin integration ?

**Comment by anna-geller** [comment]
this is just an "idea" plugin that we got from a different Slack community 

@dipeshsingh253 the plugin can extract data and store it in kestra internal storage so it can be used in downstream tasks, you can check Airbte implementation for inspiration https://docs.airbyte.com/integrations/sources/klaviyo

**Comment by anna-geller** [comment]
reminder: this is totally optional plugin open to contributors who may benefit from this plugin! we have a [bunch of other ones](https://github.com/search?q=org%3Akestra-io+label%3A%22good+first+issue%22+is%3Aopen&type=issues&utm_source=GitHub&utm_medium=github&utm_campaign=hacktoberfest2024&utm_content=Good+First+Issues) that might be more interesting and less proprietary

**Comment by dipeshsingh253** [comment]
> reminder: this is totally optional plugin open to contributors who may benefit from this plugin! we have a [bunch of other ones](https://github.com/search?q=org%3Akestra-io+label%3A%22good+first+issue%22+is%3Aopen&type=issues&utm_source=GitHub&utm_medium=github&utm_campaign=hacktoberfest2024&utm_content=Good+First+Issues) that might be more interesting and less proprietary

In this case, I would like to contribute to this issue, but first, I will explore any other good-first issues as I will have to submit 4 PRs during hacktober fest to complete it and this issue will require some time. Is it fine with you??

**Comment by MilosPaunovic** [comment]
Hi @dipeshsingh253,

Since there hasn't been any progress on the issue, I've unassigned you for now. If you decide to work on it and open a PR, let me know, and I'll be happy to reassign it to you.

In the meantime, the issue is open for anyone in the community who would like to take it on! 🚀

**Comment by babajikomali** [comment]
Hi @anna-geller @MilosPaunovic can I pick this feature?

**Comment by MilosPaunovic** [comment]
Hey @babajikomali, are there any updates on this?

**Comment by babajikomali** [comment]
Hi @MilosPaunovic, it is in progress, done with writing the tasks for profile and event endpoints.

**Comment by babajikomali** [comment]
May I know what is the purpose of trigger while creating a plugin, I am going to expose each of the important api endpoints as tasks. But not clear with the usage of triggers and conditions, can you please help with the understanding.

**Comment by MilosPaunovic** [comment]
Maybe @fdelbrayelle could help out with your question, also, we may need a separate repository for this one?

**Comment by babajikomali** [comment]
yes i guess, i checked the other plugins, each one is a repository.

**Comment by fdelbrayelle** [comment]
Hello @babajikomali 👋 
I just created the new repository [plugin-klaviyo](https://github.com/kestra-io/plugin-klaviyo), could you create a PR on it please?
Then I think you could just remove the example Trigger for now.
The important part is more to implement the [Campaign API](https://developers.klaviyo.com/en/reference/campaigns_api_overview) like @Ben8t proposed above.
Thank you for your contribution!

**Comment by fdelbrayelle** [comment]
Nevertheless if you want to learn more about triggers, [here](https://kestra.io/docs/workflow-components/triggers) are a great doc & video about them!

**Comment by babajikomali** [comment]
Sure @fdelbrayelle. Thanks for the resource.

**Comment by babajikomali** [comment]
Currently writing tasks for this flow usecase, let me know if any changes are required.
profile creation -> list creation -> adding profiles to list -> template creation -> campaign creation -> send campaign.

**Comment by babajikomali** [comment]
PR for plugin: https://github.com/kestra-io/plugin-klaviyo/pull/1
Let me know if any changes/features need to be added, I am happy to contribute.

**Comment by fdelbrayelle** [comment]
Thanks @babajikomali, we'll have a look 👀

**Comment by babajikomali** [comment]
Thanks for the review @fdelbrayelle @Malaydewangan09 , will update the code.

**Comment by babajikomali** [comment]
Having quite a hectic week, will provide the updated code by weekend.

**Comment by rishi-jat** [comment]
@fdelbrayelle Could you please assign this issue to me?
I’d like to continue the work from the previous PR and extend it further.

**Comment by fdelbrayelle** [comment]
Hey @rishi-jat 👋 
Thank you so much for your proposal.
I closed [the old PR](https://github.com/kestra-io/plugin-klaviyo/pull/1). You can re-open one by taking this code and improving it by taking the feedback from the original PR 🙏

**Comment by MilosPaunovic** [comment]
Hey @rishi-jat, are there any updates on this?

**Comment by rishi-jat** [comment]
Yes, I’m continuing the work on this. I’ve made good progress locally and will open a PR soon.

**Comment by rishi-jat** [comment]
The work is nearly complete. I’ll be raising a PR today or tomorrow.

**Comment by MilosPaunovic** [comment]
Hey @rishi-jat, is there a PR is sight?

**Comment by rishi-jat** [comment]
Hey @MilosPaunovic, most likely you’ll see the PR today. Thanks for your patience!

**Comment by rishi-jat** [comment]
CI is failing locally I am trying to fix it but facing issue should I raise PR ?

**Comment by Malaydewangan09** [comment]
Hey @rishi-jat 👋 No problem at all. Take your time fixing it, and if it is still giving you trouble, feel free to open the PR and we will help you sort it out!

**Comment by MilosPaunovic** [comment]
Hey @rishi-jat, any progress on this one?

**Comment by MilosPaunovic** [comment]
Hi @rishi-jat,

As there hasn't been any progress or updates on this, I'll unassign you for now. However, if you do open a PR, it will be reviewed, and you'll be reassigned if no one else handles it in the meantime.

Thanks for all the efforts so far! 🍀

**Comment by rishi-jat** [comment]
Make sense. I don't get much time to resolve the problem because my college exam is running. No worries i will raise PR as soon as possible.> Hi @rishi-jat,
> 
> As there hasn't been any progress or updates on this, I'll unassign you for now. However, if you do open a PR, it will be reviewed, and you'll be reassigned if no one else handles it in the meantime.
> 
> Thanks for all the efforts so far! 🍀

**Comment by MilosPaunovic** [comment]
Good luck with your exams!

**Comment by karthik7129** [comment]
Hi @MilosPaunovic I will take this up, Please assign this to me.

**Comment by fdelbrayelle** [comment]
Thanks @17297781Karthik !

**Comment by fdelbrayelle** [comment]
Thank you for your work on this @17297781Karthik 🙏 !
